### PR TITLE
Feature/dropdown support render props

### PR DESCRIPTION
**Pull request by IVLIU**
[[English Template / 英文模板](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1)]

### 🤔 这个变动的性质是？

- [x] 新特性提交
- [ ] 日常 bug 修复
- [ ] 站点、文档改进
- [ ] 演示代码改进
- [ ] 组件样式/交互改进
- [ ] TypeScript 定义更新
- [ ] 包体积优化
- [ ] 性能优化
- [ ] 功能增强
- [ ] 国际化改进
- [ ] 重构
- [ ] 代码风格优化
- [ ] 测试用例
- [ ] 分支合并
- [ ] 工作流程
- [ ] 其他改动（是关于什么的改动？）

### 💡 需求背景和解决方案

1. 使dorpdown子元素可以获取dropdown的open状态而无需受控

### 📝 更新日志

1. dropdown children支持render props形式

| 语言    | 更新描述 |
| ------- | -------- |
| 🇺🇸 英文 | dropdown children support render props form |
| 🇨🇳 中文 | dropdown children支持render props形式 |

### ☑️ 请求合并前的自查清单

⚠️ 请自检并全部**勾选全部选项**。⚠️

- [x] 文档已补充或无须补充
- [x] 代码演示已提供或无须提供
- [x] TypeScript 定义已补充或无须补充
- [x] Changelog 已提供或无须提供

---

### <samp>🤖 Generated by Copilot at 13e624d</samp>

This pull request enhances the `dropdown` component to support a function as the `children` prop and improves its open and visible behavior and style. It also fixes a dependency issue of `@types/jsdom` and adds a new demo and documentation for the `dropdown` component.

### <samp>🤖 Generated by Copilot at 13e624d</samp>

*  Add render props demo for dropdown component ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-ffd2c633c867e95b55430d9c2512a3f22d1229a389d6ab20e0350ac4abf6652aR1-R7), [link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-a956b6367b293b449799e308cce37935aee3bc03aeac117df0e4939a1f581578R1-R54), [link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-3e827e07be7d30ca7ca522ea5c430500d8946a62d213cc18cf11bf1c25c1ca97R21), [link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-ac765bc110b950888a109547694db780cc37930ca44d894616ec69ae8187b1ebR25))
  - Allow `children` prop of `dropdown.tsx` to accept a function that returns a React node ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-978b29ecd5429963a073f5adfa7ded79d88d23850853c9fb6d453a82047365beL65-R65))
  - Use merged state of `open` and `visible` props and trigger callbacks on state change ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-978b29ecd5429963a073f5adfa7ded79d88d23850853c9fb6d453a82047365beR125-R135), [link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-978b29ecd5429963a073f5adfa7ded79d88d23850853c9fb6d453a82047365beL191-L201))
  - Call `children` function with merged open state or use `children` as it is ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-978b29ecd5429963a073f5adfa7ded79d88d23850853c9fb6d453a82047365beL172-R185))
*  Improve animation of notification component ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-da4d2980e968bf7ec54db139303120e6105d73f3d214f60e9f5feec3b9c60dbdL3-R8), [link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-da4d2980e968bf7ec54db139303120e6105d73f3d214f60e9f5feec3b9c60dbdL23-R29))
  - Add `animationMaxHeight` token to control animation distance ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-da4d2980e968bf7ec54db139303120e6105d73f3d214f60e9f5feec3b9c60dbdL3-R8))
  - Use `animationMaxHeight` token in keyframes of bottom fade in animation ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-da4d2980e968bf7ec54db139303120e6105d73f3d214f60e9f5feec3b9c60dbdL23-R29))
*  Fix duplicate dependency of @types/jsdom in package.json ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L122), [link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R194))
  - Remove duplicate dependency from dependencies section ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L122))
  - Add dependency to devDependencies section ([link](https://github.com/ant-design/ant-design/pull/44931/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R194))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -192,6 +191,7 @@
     "@types/jest-environment-puppeteer": "^5.0.0",
     "@types/jest-image-snapshot": "^6.1.0",
     "@types/jquery": "^3.5.14",
+    "@types/jsdom": "^21.1.2",
```
Comment (afc163) [inline package.json:194]: 不应该修改这个

Comment (IVLIU) [reply package.json:194]: 我是从master拉的分支，大概是master和feature分支不同步吧

Comment (yoyo837) [reply package.json:194]: 那你应该从feature检出新分支

Comment (IVLIU) [reply package.json:194]: 从master拉的feature，另外我看只是位置变动，版本号没有变动，有影响吗

Comment (IVLIU) [reply package.json:194]: > 那你应该从feature检出新分支

我重新提一下

**Pull request closed without merging**